<launch>

    <arg name="plc_ip"                          default="192.168.0.43"/>
    <arg name="plc_port"                        default="50000"/>
    <arg name="xavier_ip"                       default="192.168.0.42"/>
    <arg name="xavier_port"                     default="50000"/>
    <arg name="plc_timeout"                     default="1.5"/>
    <arg name="engine_mode"                     default="0"/>
    <arg name="zero_count_encoder"              default="44377"/>
    <arg name="count_per_rotation_encoder"      default="20000"/>
    <arg name="engine_acceleration"             default="1000"/>
    <arg name="engine_jerk"                     default="10000"/>



    <node name="PLC_Connection" pkg="plc_connection" type="plc_connection_node">
    <param name="PLC_IP"                    value="$(arg plc_ip)"/>
    <param name="PLC_Port"                  value="$(arg plc_port)"/>
    <param name="PLC_Timeout"               value="$(arg plc_timeout)"/>
    <param name="Xavier_Port"               value="$(arg xavier_port)"/>
    <param name="Xavier_IP"                 value="$(arg xavier_ip)"/>
    <param name="Engine_Mode"               value="$(arg engine_mode)"/>
    <param name="ZeroCount_Encoder"         value="$(arg zero_count_encoder)"/>
    <param name="CountPerRotation_Encoder"  value="$(arg count_per_rotation_encoder)"/>
    <param name="Engine_Acceleration"       value="$(arg engine_acceleration)"/>
    <param name="Engine_Jerk"               value="$(arg engine_jerk)"/>

    <remap from="/Engine/ActualSpeed"         to="/engine/actualSpeed"/>
    <remap from="/Engine/TargetSpeed"         to="/engine/targetSpeed"/>
    <remap from="/Engine/TargetAcceleration"  to="/engine/targetAcceleration"/>
    <remap from="/Engine/TargetTorque"        to="/engine/targetTorque"/>

    <!--remap from="/Sensors/BodyAngle"           to=""/-->
    </node>
</launch>
